src/shared/FocusTrap/hooks/useFocusTrap.ts

```diff
@@ -42,11 +45,24 @@ export function useFocusTrap(
         }
     };
 
+    const setUpFocus = () => {
+        if (!elRef.current) {
+            return;
+        }
+        restoreFocusRef.current = document.activeElement;
+        const elementToFocus = getFocusableElements()[0] || elRef.current;
+        setTimeout(() => {
```

Comment:
why do we need setTimeout here?

Reply:
This is because of the dialog animation, it takes some time for the modal to be visible for us to set focus.

Reply:
Can you instead try this? Changes required in `src/shared/FocusTrap/FocusTrap.tsx` and remove setTimeout.

<img width="711" alt="image" src="https://user-images.githubusercontent.com/109717425/194549238-fbcba812-3c8a-4764-a214-568897a6dbf9.png">

Reply:
I tried the above suggestion but it didn't work. Also, I am not sure if I understand how the above is solving the transition issue.

Reply:
WRT tabIndex=-1
https://stackoverflow.com/questions/32911355/whats-the-tabindex-1-in-bootstrap-for/32912224#32912224